runtime/lua/versioncheck.lua

```diff
@@ -0,0 +1,70 @@
+local M = {}
+
+---@private
+--- Writes runtime/doc/news.txt to stdpath('state')/news.txt
+local function _write_news_to_cache()
+  local cached_news = io.open(vim.fn.stdpath('state') .. '/news.txt', 'w+')
```

Comment:
Instead of using a file, how about storing a flag in shada. If shada is disabled/empty (`nvim -n`), just skip the "show news" step.

Reply:
Not sure I understand, could you elaborate some more on that? here's how I thought about it:

We fire a 'check for news' function on CursorHold event, provided the module:

1.  hasn't been disabled (either by `vim.g.versioncheck = false`, or 
2. non-interactive session detected (e.g., `nvim -l`), or 
3. `vim.version()` has `prerelease = true`. 

'check for news' first checks if shada stored `NVIM_VERSION` exists: if it doesn't we want to store the current `vim.version()` value in it, write the current `runtime/doc/news.txt` file to `stdpath('state')..'news.txt'` and finish. I think this is where you are saying we _shouldn't_ just blindly save the current news.txt file to the stdpath('state')?

The option to show the news in the current implementation doesn't occur if `vim.g.NVIM_VERSION` doesn't exist, you're only given the option to see a diff if `NVIM_VERSION` shada does exist and it's older than current session `vim.version()` (or the corner case where the versions match, but there's somehow a different in byte size between the cached news.txt in stdpath('state') and the the runtime/doc/news.txt file).